### PR TITLE
fix(element-template-modal): support bpmn:Process and Collaboration

### LINKED ISSUE #2121: Element template plugin doesn't support non-default templates for bpmn:Process
__Describe the Bug__

The documentation (https://github.com/camunda/camunda-modeler/tree/develop/docs/element-templates#supported-bpmn-types) says that the element-templates support element bpmn:Process, but element template that doesn't have the property isDefault:true with a property appliesTo:["bpmn:Process"] is not displayed in template catalog, so it cannot be applied. Probable root cause lies in the ElementTemplatesView.getElementTemplates - for process selectedElement seems to be null.
The same template works fine for Task, so it isn't an issue of a faulty template.

__Steps to Reproduce__

1. Copy the file from https://github.com/camunda-consulting/code/blob/master/snippets/camunda-kpi-monitoring-demo/element-template/kpi.json
2. Place the element template in the resources/element-templates directory of the Camunda Modeler installation.
3. Open or restart the modeler
4. Create a BPMN diagram and click on the Open Catalog button for the process
5. The opened catalog modal window shows no templates

__Expected Behavior__

The opened modal window shows a KPI Template which can be applied to the process - elements KPI Unit and KPI Threshold are shown in the properties panel and can be set.

__Environment__

- Windows 10
- Camunda Modeler Version: 4.4.0, 4.5.0,4.6.0 (and possibly earlier versions as well)

### PR DESCRIPTION
**Pull request by MaxTru**
__Which issue does this PR address?__

Closes #2121

### CODE REVIEW

File: client/src/plugins/element-templates-modal/modeler/EditorActions.js [hunk 1]
```diff
@@ -9,50 +9,71 @@
...
+    // Register action to get the currently selected element
     editorActions.register('getSelectedElement', () => {
-      const selectedElements = selection.get();
+      return getSelectedElement();
+    });
 
-      if (selectedElements.length !== 1) {
-        return null;
+    // Register action to get the template applied to the currently selected element
+    editorActions.register('getSelectedElementAppliedElementTemplate', () => {
```
Comment (azeghers) [inline client/src/plugins/element-templates-modal/modeler/EditorActions.js:36]: I find the naming a bit heavy, but then again I'm not sure what could fit...
Maybe 'getTemplateAppliedToSelectedElement' ?

File: client/src/plugins/element-templates-modal/modeler/EditorActions.js [hunk 2]
```diff
@@ -9,50 +9,71 @@
...
+    * Get the currently selected element. If no explicit selection is made, the
+    * root element is returned. If multiple elements are selected, null is returned.
+    *
+    * @returns {Shape} selected element or root element
+    */
+    function getSelectedElement() {
       const selectedElements = selection.get();
 
-      if (selectedElements.length !== 1) {
+      if (selectedElements.length > 1 || selectedElements.length < 0) {
```
Comment (azeghers) [inline client/src/plugins/element-templates-modal/modeler/EditorActions.js:59]: What's the rationale behind this change?
I find
```if (thing !== 1)```
more readable than
```if (thing > 1 || thing < 0)```

Comment (azeghers) [reply client/src/plugins/element-templates-modal/modeler/EditorActions.js:59]: Nevermind, I read too fast, it's not the same thing, please discard

File: client/src/plugins/element-templates-modal/modeler/EditorActions.js (hunk 1 above)
Comment (azeghers) [reply client/src/plugins/element-templates-modal/modeler/EditorActions.js:36]: Ah it existed prior to this PR

**Comment by azeghers** [comment]
Looks good to me, is codecov acting out when it says a couple of lines are untested? Because as far as I understood, they are?

**Comment by nikku** [comment]
What's important is `codecov/patch` and that is going up, actually. 

Looks good to me. :+1:

**Comment by MaxTru** [comment]
The codecov hint was helpful and correct. I cleaned it up and resolved it by testing more explicit in two cases (e.g., test for `expect(xyz).to.be.null` but I had `expect(xyz).to.not.exist`)

**Pull request merged**